### ISSUE TITLE
Do not warn SA1001 (Commas should be spaced correctly) if it follows a preprocessor directive

### ISSUE THREAD

**Issue by mikernet** [DotNetAnalyzers/StyleCopAnalyzers#3816]
Example where I don't think I should need to suppress the diagnostic:
```cs
partial struct Money : IFormattable
#if !NETSTANDARD
#pragma warning disable SA1001 // Commas should be spaced correctly
    , ISpanFormattable
#pragma warning restore SA1001
#endif
```

**Comment by sharwell** [comment]
This is fine to fix. Can also likely work around:

```cs
partial struct Money :
#if !NETSTANDARD
    ISpanFormattable,
#endif
    IFormattable
```

Or:

```cs
partial struct Money
#if NET6_OR_GREATER
    : ISpanFormattable
#else
    : IFormattable
#endif
```

**Comment by mikernet** [comment]
Good point, didn't think of that tbh.
Another one, before:
```cs
public readonly struct BigDecimal : IComparable<BigDecimal>, IEquatable<BigDecimal>, IComparable, IFormattable
#if NET7_0_OR_GREATER
#pragma warning disable SA1001 // Commas should be spaced correctly
    , IFloatingPoint<BigDecimal>
#pragma warning restore SA1001
#endif
```
After:
```cs
public readonly struct BigDecimal :
#if NET7_0_OR_GREATER
    IFloatingPoint<BigDecimal>,
#endif
    IComparable<BigDecimal>, IEquatable<BigDecimal>, IComparable, IFormattable
```

That works for me.

**Comment by mikernet** [comment]
Given that there is a suitable alternative form that avoids the warning, I'm fine with this being closed if you don't think it is needed. I prefer the suggested form anyway, the leading comma bothered me haha.

**Comment by sharwell** [comment]
I'm leaving this open because it's fairly standard for us to ignore the end-of-line/beginning-of-line rules in the presence of preprocessor directives. We could either ignore all directives here, or we could just ignore conditional directives (`#if`/`#elif`/`#else`/`#endif`). I think I lean towards the latter.
